src/bridge/bridge.c

```diff
@@ -746,10 +725,34 @@ main (int argc,
       return 0;
     }
 
-  if (!opt_interactive && isatty (1))
+  if (!opt_interactive)
     {
-      g_printerr ("cockpit-bridge: no option specified\n");
-      return 2;
+      if (isatty (1))
+        {
+          g_printerr ("cockpit-bridge: no option specified\n");
+          return 2;
+        }
+
+      /* Connect stderr to the local journal explicitly.  This makes
+       * sure that we log to the machine where this process is
+       * running, and that glib always does structured logging even if
+       * our stderr was redirected to a pipe to capture errors from
+       * any wrapper that has started us (like sudo).
+       */
+
+      int fd = sd_journal_stream_fd ("cockpit/session", LOG_WARNING, 0);
+
+      /* If it didn't work, then there's no journal.  That's OK: we'll
+       * just send the output back to the client after all.
+       *
+       * If it did work, rename the fd to 2 (stderr).
+       */
+      if (fd >= 0)
```

Comment:
If we get a failure to connect to the journal, and we're relying on this redirection to prevent misinterpreting sudo's stderr messages, then this is gonna be a problem...

Reply:
The main point of connecting to journald here is to convince glib to do the usual structured logging, no?  (And in the ssh case to keep the messages on the same host as the processes.)

So if we can't connect to journald, then glib probably can't either and will fall back to unstructured logging. I think that's just fine.

For the ssh case, we could try connecting to the old school syslog thing. How is that done, actually?